marketplace/deployer_util/deploy.sh

```diff
@@ -48,6 +48,28 @@ NAMESPACE="$(/bin/print_config.py \
 export NAME
 export NAMESPACE
 
+# Kubeflow hack: Remove the owner reference on cluster-scoped IAM resources.
+set +e
+if kubectl auth can-i list,patch clusterroles | grep 'yes'; then
+  deployer_clusterroles=($(kubectl get clusterroles \
+    -l 'app.kubernetes.io/name'="$NAME" \
+    --output=custom-columns=NAME:.metadata.name \
+    | sed -n '1!p')) # removes the column header, which we have to print
```

Comment:
FYI there are two CRs created and i think this will only leave one
default:kubeflow-pipelines-2-deployer-cleanup-cr 
default:kubeflow-pipelines-2:deployerServiceAccount-r0

Verified in the cluster only 1 got removed. 
Same as the crb

Reply:
Thanks for the catch - I missed that the cleanup CR and CRB have a different component label value. Will update.

Reply:
Can you provide the output of `kubectl get clusterrole --output=json` for the CR that was left behind?

Reply:
```
{
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRoleBinding",
    "metadata": {
        "creationTimestamp": "2019-10-23T21:35:41Z",
        "labels": {
            "app.kubernetes.io/component": "deployer.marketplace.cloud.google.com",
            "app.kubernetes.io/name": "kubeflow-pipelines-2",
            "app.kubernetes.io/namespace": "default"
        },
        "name": "default:kubeflow-pipelines-2:deployerServiceAccount-rb0",
        "ownerReferences": [
            {
                "apiVersion": "app.k8s.io/v1beta1",
                "blockOwnerDeletion": true,
                "kind": "Application",
                "name": "kubeflow-pipelines-2",
                "uid": "0fca9341-f5dd-11e9-a57e-42010a80019e"
            }
        ],
        "resourceVersion": "7369",
        "selfLink": "/apis/rbac.authorization.k8s.io/v1/clusterrolebindings/default%3Akubeflow-pipelines-2%3AdeployerServiceAccount-rb0",
        "uid": "0fe09db4-f5dd-11e9-a57e-42010a80019e"
    },
    "roleRef": {
        "apiGroup": "rbac.authorization.k8s.io",
        "kind": "ClusterRole",
        "name": "default:kubeflow-pipelines-2:deployerServiceAccount-r0"
    },
    "subjects": [
        {
            "kind": "ServiceAccount",
            "name": "kubeflow-pipelines-2-deployer-sa",
            "namespace": "default"
        }
    ]
}
```

and 
```
{
    "apiVersion": "rbac.authorization.k8s.io/v1",
    "kind": "ClusterRole",
    "metadata": {
        "creationTimestamp": "2019-10-23T21:35:41Z",
        "labels": {
            "app.kubernetes.io/component": "deployer.marketplace.cloud.google.com",
            "app.kubernetes.io/name": "kubeflow-pipelines-2",
            "app.kubernetes.io/namespace": "default"
        },
        "name": "default:kubeflow-pipelines-2:deployerServiceAccount-r0",
        "ownerReferences": [
            {
                "apiVersion": "app.k8s.io/v1beta1",
                "blockOwnerDeletion": true,
                "kind": "Application",
                "name": "kubeflow-pipelines-2",
                "uid": "0fca9341-f5dd-11e9-a57e-42010a80019e"
            }
        ],
        "resourceVersion": "7372",
        "selfLink": "/apis/rbac.authorization.k8s.io/v1/clusterroles/default%3Akubeflow-pipelines-2%3AdeployerServiceAccount-r0",
        "uid": "0fe6d0cc-f5dd-11e9-a57e-42010a80019e"
    },
    "rules": [
        {
            "apiGroups": [
                "apiextensions.k8s.io"
            ],
            "resources": [
                "customresourcedefinitions"
            ],
            "verbs": [
                "*"
            ]
        }
    ]
}

```